core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2452,16 +2453,16 @@ object Stream {
     * Creates a stream that, when run, fails with the supplied exception.
     *
     * @example {{{
-    * scala> import scala.util.Try
-    * scala> Try(Stream.raiseError(new RuntimeException).toList)
-    * res0: Try[List[Nothing]] = Failure(java.lang.RuntimeException)
     * scala> import cats.effect.IO
-    * scala> Stream.raiseError(new RuntimeException).covary[IO].compile.drain.attempt.unsafeRunSync
+    * scala> Stream.raiseError[IO, Nothing](new RuntimeException).compile.drain.attempt.unsafeRunSync
     * res0: Either[Throwable,Unit] = Left(java.lang.RuntimeException)
     * }}}
     */
-  def raiseError[F[x] >: Pure[x], O](e: Throwable): Stream[F, O] =
+  def raiseError[F[x], O](e: Throwable)(
```

Comment:
When I was adding the explicit types to calls to `raiseError` I noticed that I was able to use `Nothing` as the output type pretty much everywhere. That made me wonder if the signature of this method should change to 

```scala
def raiseError[F[x]](e: Throwable)(implicit AE: ApplicativeError[F, Throwable]): Stream[F, Nothing]
```

which I think would be a little easier to deal with as a caller?

I left 77f9d0ba as a separate commit for now to show the difference, but I assume we'll either want to revert it or squash it, depending on whether sticking with `Nothing` makes sense.

Reply:
Yeah, let's remove the output param from `raiseError`.

Reply:
Implemented and squashed in 3a64d4d4